add buf beta studio-agent command with request logger (#1137)
bufbuild/buf: CHANGELOG.md

## [Unreleased]

> - Add `buf beta studio-agent` command which supports running an HTTP(S) server as the Studio agent.

Reviewer: ```suggestion
- Add `buf beta studio-agent` command to support the upcoming Buf Studio.
```

Given that "the Studio agent" doesn't say much about what its for, I think its better to note like this.
